packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -0,0 +1,819 @@
+
+    const {channelResults, outbox} = await this.takeActions(channelIds);
+
+    (outbox[0].params.data as Payload).objectives = channelIds.map(channelId => ({
+      type: 'CloseChannel',
+      participants: [],
+      data: {targetChannelId: channelId, fundingStrategy: 'Unknown'},
+    }));
+
+    return {channelResults: mergeChannelResults(channelResults), outbox: mergeOutgoing(outbox)};
+  }
+
+  async closeChannel({channelId}: CloseChannelParams): Promise<SingleChannelOutput> {
+    const {outbox, channelResults} = await this.closeChannels([channelId]);
+    return {outbox, channelResult: channelResults[0]};
+  }
+
+  private async closeChannelInternal(channelId: Bytes32): Promise<void> {
+    await this.store.lockApp(
+      channelId,
+      async (tx, channel) => {
+        if (hasSupportedState(channel) && !isMyTurn(channel))
+          throw new CloseChannel.CloseChannelError(
+            CloseChannel.CloseChannelError.reasons.notMyTurn
+          );
+
+        await this.store.addObjective(
+          {
+            type: 'CloseChannel',
+            participants: [],
+            data: {targetChannelId: channelId, fundingStrategy: channel.fundingStrategy},
+          },
+          tx
+        );
+      },
+      () => {
+        throw new CloseChannel.CloseChannelError(
+          CloseChannel.CloseChannelError.reasons.channelMissing,
+          {channelId}
+        );
+      }
+    );
+  }
+
+  async getLedgerChannels(
+    assetHolderAddress: string,
+    participants: APIParticipant[]
+  ): Promise<MultipleChannelOutput> {
+    const channelStates = await this.store.getLedgerChannels(
+      assetHolderAddress,
+      participants.map(convertToParticipant)
+    );
+    return {
+      channelResults: mergeChannelResults(channelStates.map(ChannelState.toChannelResult)),
+      outbox: [],
+    };
+  }
+
+  async getChannels(): Promise<MultipleChannelOutput> {
+    const channelStates = await this.store.getChannels();
+    return {
+      channelResults: mergeChannelResults(
+        channelStates.map(cS => ChannelState.toChannelResult(cS))
+      ),
+      outbox: [],
+    };
+  }
+
+  async getState({channelId}: GetStateParams): Promise<SingleChannelOutput> {
+    try {
+      const channel = await this.store.getChannel(channelId);
+
+      return {
+        channelResult: ChannelState.toChannelResult(channel),
+        outbox: [],
+      };
+    } catch (err) {
+      this.logger.error({err}, 'Could not get channel');
+      throw err;
+    }
+  }
+
+  async pushMessage(rawPayload: unknown): Promise<MultipleChannelOutput> {
+    const wirePayload = validatePayload(rawPayload);
+    return this._pushMessage(wirePayload).catch(e => {
+      throw new PushMessageError('Error during pushMessage', {
+        thisWalletVersion: WALLET_VERSION,
+        payloadWalletVersion: wirePayload.walletVersion,
+        cause: e,
+      });
+    });
+  }
+
+  async _pushMessage(wirePayload: WirePayload): Promise<MultipleChannelOutput> {
```

Comment:
I feel like many of these methods should be `private. But we can tackle that in separate a PR.